### ISSUE TITLE
[BUG] layoutId cannot be changed?

### ISSUE THREAD

**Issue by desiprisg** [framer/motion#1411]
** Describe the bug**

I have noticed that motion component behave as the layoutId provided to them on the first render remains the same forever, even though its passed value changes.

**IMPORTANT: Provide a CodeSandbox reproduction of the bug**

If this is something that is not intended, I will try to reproduce it in sandbox

**Steps to reproduce**

1) Initialize a state as undefined(or some string) and on the first render give it a different value.
2) Pass this variable as the layoutId in your motion component.
3) Try to transition to a different component with its layoutId being the second value.
4) The transition does not work.

**Expected behavior**

layoudId should update when passed a new value.

I'm currently setting the layoutId of an image gallery component(to the url of the image) so it would transition to the page with that same image (the image in the page has the correct layoutId). It only works for the first image of the gallery, while the others "snap" to position. If I were to add the layoutId to each image instead of the gallery itself, I would get weird behaviour with AnimatePresence since the images repeat themselves like in the example in the docs.

I would love some feedback on how layoutId works since I cannot find anything else in the docs and it is always used with a constant string like "underline". Are constant values the only ones that are acceptable?

**Comment by mattgperry** [comment]
Yeah currently it can't be changed. You could pass the same layoutId to key to try rendering it as a new component but this probably won't yield the results you want. But this is definietly a bug, we should change this at some point

**Comment by desiprisg** [comment]
Thanks for the quick response. I'm currently just doing something like
``` layoutId = {`${id}_${pageOfGallery}`}```  on the image component instead of the gallery and save the page number in a context to set it in the components I want to transition to. This way, the AnimatePresence does not fight the layout. It really would be a useful addition though since doing other stuff is getting more complex now since I have to keep track of the page all the time.

**Comment by samselikoff** [comment]
@mattgperry I'm running into something similar and curious about this sentence:

> You could pass the same layoutId to key to try rendering it as a new component but this probably won't yield the results you want.

I'm rendering two months side-by-side in a calendar (say, January and February) and there's a Next button that lets me cycle the calendars (similar to the datepicker on airbnb). I'm using `layoutId` so the right calendar (January) can animate to its new location on the left. `layoutId="jan-2022"` by itself doesn't work but if I add `key="jan-2022"` to the same component it does work.

Is this expected, or is it bad Framer Motion code?

**Comment by desiprisg** [comment]
While I haven't touched framer-motion in some time and I might be wrong, I think that you don't need `layoutId` at all for this. I'm pretty sure `layout` would suffice for your use case since this:
>If `true`, this component will automatically animate to its new position when its layout changes

sounds a lot like what you are describing. (https://www.framer.com/docs/component/###layout)

In my case, I wanted to change the `layoutId` of the component based on some action, and this is apparently not possible yet, so an option was to change the key as well so there would be a new instance so the `layoutId` would change, but this would mess up other `AnimatePresence` stuff if I remember correctly, so I found some hacky solution to make it all work. I'm pretty sure that's what @mattgperry was proposing.

I don't see you mentioning changing the `layoutId` at all, so I'm a bit confused about how this is all coded up. A codesandbox would be great!

I know the question was not directed at me but glad if I could be of help!

**Comment by qstearns** [comment]
Hello! We are currently running into this issue in our team. Is a solution for this one on the roadmap at all?

**Comment by mattgperry** [comment]
Not currently, no

**Comment by jrolfs** [comment]
I've run into this quite a few times and have usually been able to work around it, but I'd love to have more control over `layoutId` animations. @mattgperry, could you provide any guidance on where to start on this if I wanted to try to help work on it?

Thank you!!

**Comment by benface** [comment]
Looks like Jules didn't make it.

**Comment by smanzler** [comment]
I am also having this issue. Is there a reason this PR failed? Is there another way to fix this. For example, creating a new instance of create-projection-node instead of trying to change the current one or would that not work either? Just want to learn more about the issue.
